src/NHibernate/Mapping/ByCode/FakeModelExplicitDeclarationsHolder.cs

```diff
@@ -15,6 +15,8 @@ public class FakeModelExplicitDeclarationsHolder : IModelExplicitDeclarationsHol
 		private readonly IEnumerable<MemberInfo> idBags = Enumerable.Empty<MemberInfo>();
 		private readonly IEnumerable<MemberInfo> lists = Enumerable.Empty<MemberInfo>();
 		private readonly IEnumerable<MemberInfo> manyToManyRelations = Enumerable.Empty<MemberInfo>();
+		private readonly IEnumerable<MemberInfo> entityManyToManyRelations = Enumerable.Empty<MemberInfo>();
+		private readonly IEnumerable<MemberInfo> keyManyToManyRelations = Enumerable.Empty<MemberInfo>();
```

Comment:
Needed to be able to tell the difference between key based many-to-many and entity based many-to-many relationships.  I could have used only the two lists and made the ManyToManyRelations property return the union of them both, but this seemed to make the Fake simpler to use for anyone that wanted to set the ManyToManyRelations list directly in other tests.